model/task_lifecycle.go

```diff
@@ -1065,7 +1065,7 @@ func UpdateDisplayTask(t *task.Task) error {
 			hasUnfinishedTasks = true
 		}
 		// the display task's status will be the highest priority of its exec tasks
-		statuses = append(statuses, execTask.ResultStatus())
+		statuses = append(statuses, execTask.Status)
```

Comment:
If you're going to make this change, many other places need to change as well to handle the fact that the status now requires the details struct to be expressed fully (you'll also need to set that here)